src/main/java/com/mongodb/hibernate/jdbc/MongoResultSet.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    @Override
+    public long getLong(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        return getValue(columnIndex, bsonValue -> bsonValue.asInt64().longValue(), 0L);
+    }
+
+    @Override
+    public double getDouble(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        return getValue(columnIndex, bsonValue -> bsonValue.asDouble().getValue(), 0)
+                .doubleValue();
+    }
+
+    @Override
+    public byte @Nullable [] getBytes(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException();
+    }
+
+    @Override
+    public @Nullable Date getDate(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable Time getTime(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable Time getTime(int columnIndex, Calendar cal) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable Timestamp getTimestamp(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable Timestamp getTimestamp(int columnIndex, Calendar cal) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException("TODO-HIBERNATE-42 https://jira.mongodb.org/browse/HIBERNATE-42");
+    }
+
+    @Override
+    public @Nullable BigDecimal getBigDecimal(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        return getValue(
+                columnIndex,
+                bsonValue -> bsonValue.asDecimal128().decimal128Value().bigDecimalValue());
+    }
+
+    @Override
+    public @Nullable Array getArray(int columnIndex) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException();
+    }
+
+    @Override
+    public <T> @Nullable T getObject(int columnIndex, Class<T> type) throws SQLException {
+        checkClosed();
+        checkColumnIndex(columnIndex);
+        throw new FeatureNotSupportedException();
```

Comment:
The message in `MongoPreparedStatement.setObject` is `"To be implemented during Array / Struct tickets"`. Let's make them consistent.